### ISSUE TITLE
Wrong reference for SelectByShuffling

### ISSUE THREAD

**Issue by gverbock** [feature-engine/feature_engine#383]
If I am right, the calculation of the "scorer" between the unshuffled and the shuffled data set is not fair. 

The initial model performance is computed on the test set only:

```
# store initial model performance
self.initial_model_performance_ = model["test_score"].mean()
```

while the shuffled performance is computed on the full (shuffled) data set

```
# determine the performance with the shuffled feature
performance = np.mean(
            [scorer(m, X_shuffled, y) for m in model["estimator"]]
)
```

This leads to negative values for the performance drift 
```
performance_drift = self.initial_model_performance_ - performance
```

A mapping of CV model to the fold used for test could be a solution.

**Comment by solegalli** [comment]
Yes, good catch @gverbock !

Would you be able to try and fix that? If yes, please pull the branch `get_feature_names` and make a PR to that branch as well :)

Thank you!

**Comment by solegalli** [comment]
closed in #384
